### PR TITLE
add dexy-cli

### PR DESCRIPTION
**Pull request by code-for-uss**
Contracts have not been updated yet except `Emission Contract`

### CODE REVIEW

File: dexy-cli/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# dexy-cli
+
+```shell
+sbt assembly
+java -jar target/scala-2.12/dexy-test-{VERSION}.jar USER_ADRESS VALUE FEE
+```
+
+Example:
+```shell
+java -jar target/scala-2.12/dexy-test-0.0.1.jar 9get143TQtSKpf5WWEuvQD4iUp4oCCiSvPneyybn7LGWsosHNya 10000000000 1100000
```
Comment (kushti) [inline dexy-cli/README.md]: dexy-cli, not dexy-test now

**Comment by kushti** [comment]
Got error when running `dexy-cli`:

```
Started app with args
WrappedArray(9get143TQtSKpf5WWEuvQD4iUp4oCCiSvPneyybn7LGWsosHNya, 10000000000, 1100000)

Exception in thread "main" java.lang.ExceptionInInitializerError
	at utils.OnChainData$.getLastEmissionBox(OnChainData.scala:18)
	at ergoplatform.dexy.DexyOffChain$.buyDexyUSDFromEmissionBox(DexyOffChain.scala:24)
	at ergoplatform.dexy.DexyObj$.$anonfun$main$1(DexyObj.scala:16)
	at ergoplatform.dexy.DexyObj$.$anonfun$main$1$adapted(DexyObj.scala:11)
	at org.ergoplatform.appkit.RestApiErgoClient.execute(RestApiErgoClient.java:52)
	at ergoplatform.dexy.DexyObj$.main(DexyObj.scala:11)
	at ergoplatform.dexy.DexyObj.main(DexyObj.scala)
Caused by: sigmastate.lang.exceptions.TyperException: 
line 3:   val emissionNFT = emissionNFTId // to identify LP box for future use
                            ^
Cannot assign type for variable 'emissionNFTId' because it is not found in env Map(bit_& -> [T](T,T) => T, decodePoint -> (Coll[SByte$]) => SGroupElement, substConstants -> [T](Coll[SByte$],Coll[SInt$],Coll[T]) => Coll[SByte$], * -> [T](T,T) => T, bit_>>> -> [T](T,T) => T, ZKProof -> (SSigmaProp) => SBoolean, <= -> [T](T,T) => SBoolean, % -> [T](T,T) => T, fromBase58 -> (SString) => Coll[SByte$], executeFromSelfReg -> [T](SByte$,Option[T]) => T, atLeast -> (SInt$,Coll[SSigmaProp]) => SSigmaProp, anyOf -> (Coll[SBoolean]) => SBoolean, < -> [T](T,T) => SBoolean, unary_- -> [T](T) => T, executeFromVar -> [T](SByte$) => T, unary_! -> (SBoolean) => SBoolean, bit_^ -> [T](T,T) => T, fromBase64 -> (SString) => Coll[SByte$], || -> (SBoolean,SBoolean) => SBoolean, >= -> [T](T,T) => SBoolean, bit_>> -> [T](T,T) => T, min -> [T](T,T) => T, proveDlog -> (SGroupElement) => SSigmaProp, deserialize -> [T](SString) => T, sha256 -> (Coll[SByte$]) => Coll[SByte$], - -> [T](T,T) => T, sigmaProp -> (SBoolean) => SSigmaProp, bit_<< -> [T](T,T) => T, xorOf -> (Coll[SBoolean]) => SBoolean, max -> [T](T,T) => T, getVar -> [T](SByte$) => Option[T], == -> [T](T,T) => SBoolean, allZK -> (Coll[SSigmaProp]) => SSigmaProp, avlTree -> (SByte$,Coll[SByte$],SInt$,Option[SInt$]) => SAvlTree, unary_~ -> [T](T) => T, anyZK -> (Coll[SSigmaProp]) => SSigmaProp, byteArrayToBigInt -> (Coll[SByte$]) => SBigInt$, + -> [T](T,T) => T, != -> [T](T,T) => SBoolean, binary_| -> (Coll[SByte$],Coll[SByte$]) => Coll[SByte$], && -> (SBoolean,SBoolean) => SBoolean, waitingPeriod -> SInt$, bit_| -> [T](T,T) => T, proveDHTuple -> (SGroupElement,SGroupElement,SGroupElement,SGroupElement) => SSigmaProp, byteArrayToLong -> (Coll[SByte$]) => SLong$, longToByteArray -> (SLong$) => Coll[SByte$], outerJoin -> [K,L,R,O](Coll[(K,L)],Coll[(K,R)],(K,L) => O,(K,R) => O,(K,L,R) => O) => Coll[(K,O)], ^ -> (SBoolean,SBoolean) => SBoolean, / -> [T](T,T) => T, > -> [T](T,T) => SBoolean, allOf -> (Coll[SBoolean]) => SBoolean, blake2b256 -> (Coll[SByte$]) => Coll[SByte$])
	at sigmastate.lang.SigmaTyper$.error(SigmaTyper.scala:713)
	at sigmastate.lang.SigmaTyper.assignType(SigmaTyper.scala:82)
	at sigmastate.lang.SigmaTyper.$anonfun$assignType$2(SigmaTyper.scala:57)
	at scala.collection.TraversableLike$WithFilter.$anonfun$foreach$1(TraversableLike.scala:877)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:876)
	at sigmastate.lang.SigmaTyper.assignType(SigmaTyper.scala:55)
	at sigmastate.lang.SigmaTyper.typecheck(SigmaTyper.scala:607)
	at sigmastate.lang.SigmaCompiler.typecheck(SigmaCompiler.scala:46)
	at sigmastate.lang.SigmaCompiler.typecheck(SigmaCompiler.scala:52)
	at org.ergoplatform.ErgoScriptPredef$.compileWithCosting(ErgoScriptPredef.scala:23)
	at org.ergoplatform.appkit.JavaHelpers$.compile(JavaHelpers.scala:318)
	at org.ergoplatform.appkit.JavaHelpers.compile(JavaHelpers.scala)
	at org.ergoplatform.appkit.impl.ErgoScriptContract.getErgoTree(ErgoScriptContract.java:48)
	at ergoplatform.dexy.DexyContracts$.swappingErgoTree$lzycompute$1(DexyContracts.scala:350)
	at ergoplatform.dexy.DexyContracts$.swappingErgoTree$1(DexyContracts.scala:350)
	at ergoplatform.dexy.DexyContracts$.swappingAddress$lzycompute$1(DexyContracts.scala:351)
	at ergoplatform.dexy.DexyContracts$.swappingAddress$1(DexyContracts.scala:351)
	at ergoplatform.dexy.DexyContracts$.$anonfun$new$1(DexyContracts.scala:353)
	at ergoplatform.dexy.DexyContracts$.$anonfun$new$1$adapted(DexyContracts.scala:309)
	at org.ergoplatform.appkit.RestApiErgoClient.execute(RestApiErgoClient.java:52)
	at ergoplatform.dexy.DexyContracts$.<init>(DexyContracts.scala:309)
	at ergoplatform.dexy.DexyContracts$.<clinit>(DexyContracts.scala)
	... 7 more
```

File: dexy-cli/README.md (hunk 1 above)
Comment (code-for-uss) [reply dexy-cli/README.md]: Fixed

**Comment by code-for-uss** [comment]
> Got error when running `dexy-cli`:
> 
> ```
> Started app with args
> WrappedArray(9get143TQtSKpf5WWEuvQD4iUp4oCCiSvPneyybn7LGWsosHNya, 10000000000, 1100000)
> 
> Exception in thread "main" java.lang.ExceptionInInitializerError
> 	at utils.OnChainData$.getLastEmissionBox(OnChainData.scala:18)
> 	at ergoplatform.dexy.DexyOffChain$.buyDexyUSDFromEmissionBox(DexyOffChain.scala:24)
> 	at ergoplatform.dexy.DexyObj$.$anonfun$main$1(DexyObj.scala:16)
> 	at ergoplatform.dexy.DexyObj$.$anonfun$main$1$adapted(DexyObj.scala:11)
> 	at org.ergoplatform.appkit.RestApiErgoClient.execute(RestApiErgoClient.java:52)
> 	at ergoplatform.dexy.DexyObj$.main(DexyObj.scala:11)
> 	at ergoplatform.dexy.DexyObj.main(DexyObj.scala)
> Caused by: sigmastate.lang.exceptions.TyperException: 
> line 3:   val emissionNFT = emissionNFTId // to identify LP box for future use
>                             ^
> Cannot assign type for variable 'emissionNFTId' because it is not found in env Map(bit_& -> [T](T,T) => T, decodePoint -> (Coll[SByte$]) => SGroupElement, substConstants -> [T](Coll[SByte$],Coll[SInt$],Coll[T]) => Coll[SByte$], * -> [T](T,T) => T, bit_>>> -> [T](T,T) => T, ZKProof -> (SSigmaProp) => SBoolean, <= -> [T](T,T) => SBoolean, % -> [T](T,T) => T, fromBase58 -> (SString) => Coll[SByte$], executeFromSelfReg -> [T](SByte$,Option[T]) => T, atLeast -> (SInt$,Coll[SSigmaProp]) => SSigmaProp, anyOf -> (Coll[SBoolean]) => SBoolean, < -> [T](T,T) => SBoolean, unary_- -> [T](T) => T, executeFromVar -> [T](SByte$) => T, unary_! -> (SBoolean) => SBoolean, bit_^ -> [T](T,T) => T, fromBase64 -> (SString) => Coll[SByte$], || -> (SBoolean,SBoolean) => SBoolean, >= -> [T](T,T) => SBoolean, bit_>> -> [T](T,T) => T, min -> [T](T,T) => T, proveDlog -> (SGroupElement) => SSigmaProp, deserialize -> [T](SString) => T, sha256 -> (Coll[SByte$]) => Coll[SByte$], - -> [T](T,T) => T, sigmaProp -> (SBoolean) => SSigmaProp, bit_<< -> [T](T,T) => T, xorOf -> (Coll[SBoolean]) => SBoolean, max -> [T](T,T) => T, getVar -> [T](SByte$) => Option[T], == -> [T](T,T) => SBoolean, allZK -> (Coll[SSigmaProp]) => SSigmaProp, avlTree -> (SByte$,Coll[SByte$],SInt$,Option[SInt$]) => SAvlTree, unary_~ -> [T](T) => T, anyZK -> (Coll[SSigmaProp]) => SSigmaProp, byteArrayToBigInt -> (Coll[SByte$]) => SBigInt$, + -> [T](T,T) => T, != -> [T](T,T) => SBoolean, binary_| -> (Coll[SByte$],Coll[SByte$]) => Coll[SByte$], && -> (SBoolean,SBoolean) => SBoolean, waitingPeriod -> SInt$, bit_| -> [T](T,T) => T, proveDHTuple -> (SGroupElement,SGroupElement,SGroupElement,SGroupElement) => SSigmaProp, byteArrayToLong -> (Coll[SByte$]) => SLong$, longToByteArray -> (SLong$) => Coll[SByte$], outerJoin -> [K,L,R,O](Coll[(K,L)],Coll[(K,R)],(K,L) => O,(K,R) => O,(K,L,R) => O) => Coll[(K,O)], ^ -> (SBoolean,SBoolean) => SBoolean, / -> [T](T,T) => T, > -> [T](T,T) => SBoolean, allOf -> (Coll[SBoolean]) => SBoolean, blake2b256 -> (Coll[SByte$]) => Coll[SByte$])
> 	at sigmastate.lang.SigmaTyper$.error(SigmaTyper.scala:713)
> 	at sigmastate.lang.SigmaTyper.assignType(SigmaTyper.scala:82)
> 	at sigmastate.lang.SigmaTyper.$anonfun$assignType$2(SigmaTyper.scala:57)
> 	at scala.collection.TraversableLike$WithFilter.$anonfun$foreach$1(TraversableLike.scala:877)
> 	at scala.collection.immutable.List.foreach(List.scala:392)
> 	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:876)
> 	at sigmastate.lang.SigmaTyper.assignType(SigmaTyper.scala:55)
> 	at sigmastate.lang.SigmaTyper.typecheck(SigmaTyper.scala:607)
> 	at sigmastate.lang.SigmaCompiler.typecheck(SigmaCompiler.scala:46)
> 	at sigmastate.lang.SigmaCompiler.typecheck(SigmaCompiler.scala:52)
> 	at org.ergoplatform.ErgoScriptPredef$.compileWithCosting(ErgoScriptPredef.scala:23)
> 	at org.ergoplatform.appkit.JavaHelpers$.compile(JavaHelpers.scala:318)
> 	at org.ergoplatform.appkit.JavaHelpers.compile(JavaHelpers.scala)
> 	at org.ergoplatform.appkit.impl.ErgoScriptContract.getErgoTree(ErgoScriptContract.java:48)
> 	at ergoplatform.dexy.DexyContracts$.swappingErgoTree$lzycompute$1(DexyContracts.scala:350)
> 	at ergoplatform.dexy.DexyContracts$.swappingErgoTree$1(DexyContracts.scala:350)
> 	at ergoplatform.dexy.DexyContracts$.swappingAddress$lzycompute$1(DexyContracts.scala:351)
> 	at ergoplatform.dexy.DexyContracts$.swappingAddress$1(DexyContracts.scala:351)
> 	at ergoplatform.dexy.DexyContracts$.$anonfun$new$1(DexyContracts.scala:353)
> 	at ergoplatform.dexy.DexyContracts$.$anonfun$new$1$adapted(DexyContracts.scala:309)
> 	at org.ergoplatform.appkit.RestApiErgoClient.execute(RestApiErgoClient.java:52)
> 	at ergoplatform.dexy.DexyContracts$.<init>(DexyContracts.scala:309)
> 	at ergoplatform.dexy.DexyContracts$.<clinit>(DexyContracts.scala)
> 	... 7 more
> ```

Fixed contract compiling

**Pull request closed without merging**